test/metrics/query.go

```diff
@@ -0,0 +1,260 @@
+	if err != nil {
+		return "", err
+	}
+
+	var bearerToken string
+	secrets, err := client.CoreV1().Secrets("openshift-monitoring").List(metav1.ListOptions{})
+	if err != nil {
+		return "", fmt.Errorf("could not list secrets in openshift-monitoring namespace")
+	}
+	for _, secret := range secrets.Items {
+		if secret.Type != corev1.SecretTypeServiceAccountToken {
+			continue
+		}
+		if !strings.HasPrefix(secret.Name, "prometheus-") {
+			continue
+		}
+		bearerToken = string(secret.Data[corev1.ServiceAccountTokenKey])
+		break
+	}
+	if len(bearerToken) == 0 {
+		return "", fmt.Errorf("error getting bearer token from prometheus service account")
+	}
+	return bearerToken, nil
+}
+
+type prometheusResponse struct {
+	Status string                 `json:"status"`
+	Data   prometheusResponseData `json:"data"`
+}
+
+type prometheusResponseData struct {
+	ResultType string       `json:"resultType"`
+	Result     model.Vector `json:"result"`
+}
+
+// CreateAnnotatedCABundleConfigMap results in configmap injected with CA bundle issued by service-ca-operator
+func CreateAnnotatedCABundleInjectionConfigMap(client *kubernetes.Clientset, configMapName, nsName string) error {
+	_, err := client.CoreV1().ConfigMaps(nsName).Create(&corev1.ConfigMap{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: configMapName,
+			Annotations: map[string]string{
+				serviceCAInjectAnnotation: "true",
+			},
+		},
+	})
+	// confirm ca bundle is injected
+	expectedDataSize := 1
+	err = wait.PollImmediate(time.Second*1, time.Second*30, func() (bool, error) {
+		configMap, err := client.CoreV1().ConfigMaps(nsName).Get(configMapName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		if len(configMap.Data) != expectedDataSize {
+			return false, fmt.Errorf("expected data size %d, got %d", expectedDataSize, len(configMap.Data))
+		}
+		return true, nil
+	})
+	return err
+}
+
+// RunQueryFromPod checks that query returns non-empty result, errors if query returns empty, and returns metrics
+// example: query := `ALERTS{alertstate="pending",alertname="PodDisruptionBudgetAtLimit",severity="warning"} == 1`
+func RunQueryFromPod(kubeconfig, kubectl, name, query, bearerToken string) (model.Vector, error) {
+	var metrics model.Vector
+	promURL := fmt.Sprintf("%s/api/v1/query?%s", prometheusBaseURL, (url.Values{"query": []string{query}}).Encode())
+	cmd := fmt.Sprintf("curl -s -H 'Authorization: Bearer %s' %q", bearerToken, promURL)
+	wait.PollImmediate(time.Second*1, time.Second*180, func() (bool, error) {
+		output, err := runExecCmd(kubeconfig, kubectl, name, cmd)
+		if err != nil {
+			return false, fmt.Errorf("curl command failed: %v\n%s", err, output)
+		}
+		result := prometheusResponse{}
+		json.Unmarshal([]byte(output), &result)
+		metrics = result.Data.Result
+		if len(metrics) == 0 {
+			return false, nil
+		}
+		return true, nil
+	})
+	if len(metrics) == 0 {
```

Comment:
Analysing whether output is correct or not should be left to the caller.

Reply:
updated, thanks